lib/screens/product_edit_screen.dart

```diff
@@ -63,167 +70,369 @@ class _TBProductEditScreenState extends State<TBProductEditScreen> {
-                                  onPressed: () {
-                                    Navigator.of(context).pop();
-                                  },
-                                  message:
-                                      "Something went wrong while editing product!"),
-                            );
-                          }
-                        },
+                              const SizedBox(width: 16),
+                              Expanded(
+                                child: TBTextInput(
+                                    textEditingController:
+                                        _priceTextEditingController,
+                                    label: "Price",
+                                    suffixText: "\$",
+                                    validator: textInputValidator),
+                              ),
+                            ],
+                          ),
+                          SizedBox(
+                              height: TBDimensions
+                                  .productEditDetailsScreen.contentSpacing),
+                          TBButton(
+                            isLoading: true,
+                            text: "Save changes",
+                            type: TBButtonType.filled,
+                            onPressed: () {
+                              if (_formKey.currentState!.validate()) {
+                                ref
+                                    .read(productProvider(value.id).notifier)
+                                    .edit(
+                                      TBProductModel(
+                                        id: value.id,
+                                        name: _nameTextEditingController.text
+                                            .trim(),
+                                        company: _companySelection!,
+                                        category: _categorySelection!,
+                                        description:
+                                            _descriptionTextEditingController
+                                                .text
+                                                .trim(),
+                                        discount: double.parse(
+                                            _discountTextEditingController.text
+                                                .trim()),
+                                        price: double.parse(
+                                            _priceTextEditingController.text
+                                                .trim()),
+                                        image: value.image,
+                                        onStock: value.onStock,
+                                        rating: value.rating,
+                                      ),
+                                    )
+                                    .then((value) {
+                                  showDialog(
```

Comment:
do not use then, use ref.listen() to listen if the state is successful then show the dialog. that means that edit cant be in productProvider